### ISSUE TITLE
Single shared photo is not visible in Memories

### ISSUE THREAD

**Issue by kvn61** [pulsejet/memories#457]
With shared folders all ok.

**Comment by frederikb96** [comment]
Still a problem for me, single shared photos (doesnt matter if full permissions or read-only) do not appear in timeline or folder view. I tried to place the photo in multiple locations, like directly in the timeline path or in subfolders in timeline path, but no change.
However, as said, if sharing a folder which contains photos, those are visible in timeline and folder view.

**Comment by frederikb96** [comment]
If I find time, I can look into this problem. Do you have any hint, maybe, as to which part of the code I should look first in?

**Comment by frederikb96** [comment]
I started debugging this problem. What I found out so far:
- The problem is not in the indexing process. I debugged index.php, which is regularly called by the cron job. And in index.php, shared files are correctly indexed via [processFile Function](https://github.com/pulsejet/memories/blob/c8689351e72c997fcc7e7a2270ada3043e22eae5/lib/Db/TimelineWrite.php#L41) from the db/TimelineWriter.php file.
- So I will next look into the other TimeLine files and the FSManager, which is, I guess, called when the timeline is created. So maybe it is during runtime, when the single shared files are somehow excluded.

**Comment by pulsejet** [comment]
You're right, the FSManager simply doesn't include individual file mounts right now. Only the folders are passed to the CTE and are traversed recursively.

**Comment by frederikb96** [comment]
Thanks for the info :)

I tried investigating further, but it is quite difficult for me to understand everything. If I understand it correctly, I have to:
1. Extend TimelineRoot Class: Modify the `TimelineRoot` class to include arrays that can hold information about individual files, in addition to folders.

2. Populate TimelineRoot with Shared Files: Update the logic in `FSManager`  to add shared files to these new arrays. Store information like file ids and paths in `TimelineRoot`.

3. Investigate and Modify Queries in `TimelineQuery*` Files: Somehow extend the query constructions in the `TimelineQuery*` files to handle the new arrays with files. 

I will see how far I will come with this :smile:
